### PR TITLE
fixing the issue with container reuse for consecutive invocations

### PR DESCRIPTION
**Pull request by ddragosd**
Fixes https://github.com/apache/incubator-openwhisk/issues/2788 with Docker for Mac and docker-compose.

Inspired from https://github.com/apache/incubator-openwhisk/pull/2793

Changelist:
- [x] Mount `runc` binaries from the host machine
- [x] Upgrade to `v3` docker-compose file format 
- [x] Invoker image is staretd with `privileged`, `pid:host` and `userns_mode:host`

### CODE REVIEW

**Comment by ddragosd** [comment]
@alexkli ☝️

**Comment by ddragosd** [comment]
marking this as `WIP` b/c `docker`/`runc` binaries should be detected differently on MacOSX vs other OSs that don't use a VM to run Docker inside.

**Comment by ddragosd** [comment]
as per the latest build failure, the logs show:
```[2017-09-27T18:23:32.164Z] [ERROR] [#tid_5] [DockerClientWithFileAccess] code: 127, stdout: , stderr: /usr/bin/docker: error while loading shared libraries: libltdl.so.7: cannot open shared object file: No such file or directory [marker:invoker_docker.run_error:296:10] ```

**Comment by ddragosd** [comment]
mounting only the `runc` binary fixed the travis build as well.

**Comment by alexkli** [comment]
👍

Verified this fixes the issue for me (compared to 192197e4d912d6e6c9d51f6fd0b7762439538576 that I was using before), and containers get reused.

Used the following action provided by @ddragosd to verify. It will output how often the action was reused by leveraging a global counter in nodejs:
```javascript
function main(params) {
    var name = params.name || params.payload || 'stranger';
    var place = params.place || 'somewhere';
    global._reused_counter = global._reused_counter || 0;
    global._reused_counter++;
    console.log(`Reused counter=${global._reused_counter}`);
    return {payload:  'Hello, ' + name + ' from ' + place + ' !', reused: global._reused_counter};
}
```

File: docker-compose/Makefile [hunk 1]
```diff
@@ -6,10 +6,18 @@ WSK_CLI ?= $(PROJECT_HOME)/bin/wsk
 
 OPEN_WHISK_DB_PREFIX ?= local_
 
+DOCKER_KERNEL ?= $(shell docker version --format "{{.Server.KernelVersion}}")
+RUNC_BINARY   ?= $(shell if [[ $(DOCKER_KERNEL) == *-moby || $(DOCKER_KERNEL) ==  *-boot2docker ]]; then (docker run --rm --privileged --pid=host debian nsenter -t 1 -m -u -n -i sh -c "which runc || which docker-runc"); else (which runc || which docker-runc); fi)
+DOCKER_BINARY ?= $(shell if [[ $(DOCKER_KERNEL) == *-moby || $(DOCKER_KERNEL) ==  *-boot2docker ]]; then (docker run --rm --privileged --pid=host debian nsenter -t 1 -m -u -n -i sh -c "which docker"); else (which docker); fi)
```
Comment (tysonnorris) [inline docker-compose/Makefile]: maybe add a comment on known envs where this works: Docker for Mac, boot2docker, moby etc

File: docker-compose/Makefile [hunk 2]
```diff
@@ -6,10 +6,18 @@ WSK_CLI ?= $(PROJECT_HOME)/bin/wsk
...
+
 ifndef VERBOSE
 .SILENT:
 endif
 
+.PHONY: ddd
+ddd:
+	$(shell cat ~/tmp/openwhisk/local.env) && echo $${DOCKER_IMAGE_PREFIX}
+
```
[the remark this answers is not in the document]
Comment (ddragosd) [reply docker-compose/Makefile]: good catch 😄

File: docker-compose/Makefile (hunk 1 above)
Comment (ddragosd) [reply docker-compose/Makefile]: @alexkli updated the README on #54 specifying that on Mac, Docker for Mac is required. Is that what you were thinking too ?

**Pull request merged**